sqldelight-gradle-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/GenerateSchemaTask.kt

```diff
@@ -38,7 +38,7 @@ abstract class GenerateSchemaTask : SourceTask() {
   var outputDirectory: File? = null
 
   @Internal lateinit var sourceFolders: Iterable<File>
-  @Internal @Input lateinit var properties: SqlDelightDatabaseProperties
+  @Internal lateinit var properties: SqlDelightDatabaseProperties
```

Comment:
This doesn't seem right. Also it seems like `sourceFolders` needs to be an input...

Reply:
gradle doesn't allow @Internal @Input anymore

/shrug

Reply:
I believe this means if you change you DB name, package, or folders and re-run a build the task will not be invalidated. Even worse, if you have a Gradle cache it will always pull the task from the cache and you'll have to purge the cache in order to see your changes reflected.

Reply:
i c, probably better to make them public input then

Reply:
everything i try gradle fails at me. This task persistence api stuff is garbo

Reply:
Yep it's pretty awful. Basically doing the wrong thing is easy and you have to try really, really, really hard to do the right thing, and nothing even validates when you do.